### PR TITLE
[WIP][INTLY-4048] Change theme. Add support for 3scale and SSO authentication

### PR DESCRIPTION
**Pull request by evanshortiss**
## Motivation
Motivated by the broader Solution Pattern cohesion effort described in INTLY-4045

## Why
We're working on an effort to improve the default Solution Patterns (walkthroughs) to have a common thread. This means they can be built upon to create a larger solution instead of each having to be completely standalone.

## How
* Redesigned the UI using PatternFly components and a new car parts ordering theme.
* Added the ability to enable SSO using Keycloak if `KEYCLOAK_CONFIG` is a defined environment variable set to a valid Keycloak OIDC JSON
* Added ability to authenticate using a 3scale `x-3scale-proxy-secret-token` header if the `THREESCALE_SECRET` is set
* Added session management so users can revisit the portal and see order history
* Refactored code into config, middlewares, amq, etc. modules 

## Verification Steps
* Install the walkthroughs from [this PR](https://github.com/integr8ly/tutorial-web-app-walkthroughs/pull/111) via `oc patch webapp tutorial-web-app-operator -n openshift-webapp --type=merge -p '{ "spec": { "template": { "parameters": { "WALKTHROUGH_LOCATIONS": "https://github.com/evanshortiss/tutorial-web-app-walkthroughs#INTLY-4050" }}}}'`

## Checklist:

- [ ] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member

## Progress

- [x] Coding work
- [ ] Publishing new images (referenced in WT templates JSON) to quay.io
- [ ] Land linked PRs, e.g the walkthrough PR

## Additional Notes

Note the requirements above before merging this.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,4 +2,8 @@ FROM node:10-alpine
 ADD . ./messaging-app
 WORKDIR ./messaging-app/frontend
 RUN npm i
+EXPOSE 8080
+# Remove dev/build dependencies to reduce container size
+RUN npm prune --production
+USER 1001
```
Comment (evanshortiss) [inline Dockerfile]: Might need a review on this one? OCP container permissions stuff.

File: frontend/app.js [hunk 2]
```diff
@@ -18,94 +18,86 @@
...
+const exphbs = require('express-handlebars')
+const log = require('./lib/log')
+const { getUsernameFromRequest } = require('./lib/utils')
+const { getKeycloakInstance, getSessionMiddleware, threescaleApiMiddleware } = require('./lib/middleware')
+
+const app = express()
+const keycloak = getKeycloakInstance()
+
+// Required when running behind a load balancer, e.g HAProxy
+app.set('trust proxy', true)
```
Comment (evanshortiss) [inline frontend/app.js:35]: Required for SSO redirect URIs (when using Keycloak) to be constructed correctly.

**Comment by evanshortiss** [comment]
@pb82 @aidenkeating would be great to get your eyes on this.

**Review by pb82 — APPROVED** [review]
verified with walkthrough 1b on https://tutorial-web-app-webapp.apps.eshortis-9b03.open.redhat.com/

**Comment by evanshortiss** [comment]
@pb82 image pushed - https://quay.io/repository/integreatly/rhmi-lab-nodejs-order-frontend?tab=tags

**Comment by evanshortiss** [comment]
@pb82 @david-martin @aidenkeating should we go ahead and merge this since it's in the 1.6 release?

**Comment by david-martin** [comment]
@evanshortiss I'm good to approve and merge this if it's not going to affect any production clusters.
Can you confirm 'master' tag is not referenced anywhere, and a separate tag will be created after for the 1.6 release?

**Comment by evanshortiss** [comment]
@david-martin the work I've done is integrated with the Ansible installer specific tags of images found here. I don't believe the repo is referenced directly anywhere.

* https://quay.io/repository/integreatly/rhmi-lab-nodejs-order-frontend
* https://quay.io/repository/integreatly/rhmi-lab-internal-fuse-api
* https://quay.io/repository/integreatly/rhmi-lab-order-management-ui

**Pull request merged**